pkg/lib/ownerutil/util.go

```diff
@@ -123,6 +123,33 @@ func Adoptable(target Owner, owners []metav1.OwnerReference) bool {
 	return false
 }
 
+// APIServiceAdoptable checks whether a resource with the given set of OwnerReferences
+// is "adoptable" by the target OwnerReference. This function returns true if there
+// exists an element in owners referencing the same kind target does and one of
+// the owners is in the new CSV's replaces chain, otherwise it returns false.
+func APIServiceAdoptable(replaceCSV string, target Owner, owners []metav1.OwnerReference) bool {
+	if len(owners) == 0 {
+		// Resources with no owners are not adoptable
+		return false
+	} else if len(owners) == 1 && owners[0].Name == target.GetName() {
```

Comment:
What happens when I try to deploy the same CSV in two different namespaces? A different resource of the same name? Could we get away with checking Kinds and UIDs here to be safe?